### PR TITLE
Add GetBestTFDBetweenMolecules()

### PR DESCRIPTION
**Pull request by greglandrum**
This makes it easier (and much, much faster) to compare a single conformer on one molecule with multiple conformers on another and to get the best result.

There's a bit of minor general cleanup in here as well

### CODE REVIEW

**Comment by greglandrum** [comment]
@sriniker if you have time to look at this it would be great to get your feedback

File: rdkit/Chem/TorsionFingerprints.py [hunk 1]
```diff
@@ -611,11 +609,54 @@ def GetTFDBetweenMolecules(mol1, mol2, confId1=-1, confId2=-1, useWeights=True,
...
+  # first molecule
+  torsion1 = CalculateTorsionAngles(mol1, tl, tlr, confId=confId1)
+  if useWeights:
+    weights = CalculateTorsionWeights(mol1, ignoreColinearBonds=ignoreColinearBonds)
+  best = 1e8
+  for conf in mol2.GetConformers():
+    # second molecule
+    torsion2 = CalculateTorsionAngles(mol2, tl, tlr, confId=conf.GetId())
+    if not useWeights:
+      weights = None
```
Comment (sriniker) [inline rdkit/Chem/TorsionFingerprints.py:654]: Maybe I overlook something, but why is this inside the loop and not outside (directly after "if useWeights")?

**Review by sriniker — CHANGES_REQUESTED** [review]
Looks good otherwise

File: rdkit/Chem/TorsionFingerprints.py (hunk 1 above)
Comment (greglandrum) [reply rdkit/Chem/TorsionFingerprints.py]: Good catch!

**Comment by greglandrum** [comment]
Merging after @sriniker's review

**Pull request merged**